src/main/kotlin/at/bitfire/dav4jvm/Property.kt

```diff
@@ -7,17 +7,19 @@
 package at.bitfire.dav4jvm
 
 import at.bitfire.dav4jvm.exception.InvalidPropertyException
-import org.xmlpull.v1.XmlPullParser
 import java.io.Serializable
 import java.util.LinkedList
 import java.util.logging.Level
 import java.util.logging.Logger
+import org.xmlpull.v1.XmlPullParser
 
 /**
  * Represents a WebDAV property.
  *
  * Every [Property] must define a static field (use `@JvmStatic`) called `NAME` of type [Property.Name],
  * which will be accessed by reflection.
+ *
+ * Every [Property] must be a data class.
```

Comment:
_Must_ is a very strong requirement.

Maybe _should_, plus an explanation why (we want to be able to compare the properties and have `toString` for debugging).